packages/core/src/CorePlugin.ts

```diff
@@ -86,11 +85,11 @@ export class CorePlugin extends JWPlugin {
      * Delete in the backward direction (backspace key expected behavior).
      */
     deleteBackward(): void {
-        const range = this.editor.vDocument.range;
+        const range = this.editor.vDocument.selection.range;
```

Comment:
Commands work on *a* range, not *the* range. It is the command responsibility to decide which is the default range it uses when none is provided. Some commands do not need the range so setting a default value for `params.range` through `execCommand` would make it weird to have range in their params if they don't use it, plus it would be impossible to detect whether a range was provided or not. Maybe I want to write a command which does something at the start of the document unless you give it another range to work on ? Or maybe I want to do something on the first heading node ? So I could write:
```
const nodes = params.range ? params.range.selectedNodes() : this.editor.vDocument.root.firstDescendant(h1);
```
I think a more obvious issue is the fact that `this.editor.vDocument` is a pain to read and write and that might be part of why you are skeptical of the `this.editor.vDocument.selection.range`. I agree that something might be done to alleviate that issue.